stdlib/public/runtime/ImageInspectionStatic.cpp

```diff
@@ -0,0 +1,54 @@
+//===-- ImageInspectionStatic.cpp -------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Implementation of functions to read data sections from static executable.
+//
+//===----------------------------------------------------------------------===//
+
+#include "ImageInspection.h"
+
+// Currently only tested on linux but should work for any ELF platform
+#if defined(__ELF__) && defined(__linux__)
+
+// These are defined in swift_sections.S
+// Used in swift_sections.S to mark the start of a section
+struct SectionInfo {
+  uint64_t size;
+  const uint8_t data[0];
+};
+extern const SectionInfo __swift2_protocol_conformances_start;
```

Comment:
You ought to be able to preserve the existing dotted symbol names by declaring these with an `asm` attribute, e.g. `extern const SectionInfo ProtocolConformancesStart asm(".swift2_protocol_conformances_start");`, which would also give the declarations a slightly friendlier source name. I'm a bit concerned about the potential of other users outside of the compiler depending on the existing symbol names.